src/main/kotlin/com/terraformation/backend/tracking/db/T0PlotStore.kt

```diff
@@ -251,6 +252,65 @@ class T0PlotStore(
     )
   }
 
+  fun assignT0TempZoneSpeciesDensities(
+      plantingZoneId: PlantingZoneId,
+      densities: List<SpeciesDensityModel>,
+  ) {
+    requirePermissions { updateT0(plantingZoneId) }
+
+    val now = clock.instant()
+    val currentUserId = currentUser().userId
+
+    dslContext.transaction { _ ->
+      with(PLANTING_ZONE_T0_TEMP_DENSITIES) {
+        dslContext
+            .deleteFrom(this)
+            .where(PLANTING_ZONE_ID.eq(plantingZoneId))
+            .and(SPECIES_ID.notIn(densities.map { it.speciesId }))
+            .execute()
+
+        var insertQuery =
+            dslContext.insertInto(
+                this,
```

Comment:
This is far enough away from the `with` that I think it'd be clearer to use the table name here.